programs/ziti-edge-tunnel/process_cmd.c

```diff
@@ -207,55 +248,109 @@ bool process_tunnel_commands(const tunnel_command *tnl_cmd, command_cb cb, void
 
         case TunnelCommand_AddIdentity : {
             cmd_accepted = true;
-            tunnel_add_identity tunnel_add_identity_cmd = {0};
-            if (tnl_cmd->data == NULL ||
-                parse_tunnel_add_identity(&tunnel_add_identity_cmd, tnl_cmd->data, strlen(tnl_cmd->data)) < 0 ||
-                (tunnel_add_identity_cmd.jwtFileName == NULL && tunnel_add_identity_cmd.jwtContent == NULL)) {
+            if (tnl_cmd->data == NULL) {
                 result.error = "invalid command";
                 result.success = false;
-                free_tunnel_add_identity(&tunnel_add_identity_cmd);
                 break;
             }
 
-            if (config_dir == NULL) {
-                result.error = "config directory not set";
+            if (!uses_config_dir) {
+                result.error = "config directory not set, add command requires an identity-dir";
                 result.success = false;
                 break;
             }
 
-            if (tunnel_add_identity_cmd.jwtFileName == NULL) {
-                result.error = "identity filename not provided";
+            tunnel_add_identity tunnel_add_identity_cmd = {0};
+            int parse_result = parse_tunnel_add_identity(&tunnel_add_identity_cmd, tnl_cmd->data, strlen(tnl_cmd->data));
+
+            if (parse_result < 0) {
+                result.error = "invalid command - could not parse";
                 result.success = false;
+                free_tunnel_add_identity(&tunnel_add_identity_cmd);
                 break;
             }
+            char *null = NULL;
+            bool is_jwt = tunnel_add_identity_cmd.jwtFileName != null || tunnel_add_identity_cmd.jwtContent != null;
+            bool is_url = tunnel_add_identity_cmd.controllerURL != null;
+            bool is_3rd_party_ca = tunnel_add_identity_cmd.cert != null || tunnel_add_identity_cmd.key != null;
+            int enrollment_methods_supplied = (is_jwt + is_url + is_3rd_party_ca);
```

Comment:
this would be clearer (and I think safer) if it was a bool:
```
bool enrollment_methods_supplied = is_jwt || is_url || is_3rd_party_ca;
```

Reply:
first implementation needed it, but new/better impl doesn't. will make this change. thx